test/jdk/com/sun/security/auth/callback/TextCallbackHandler/Default.java

```diff
@@ -23,31 +23,60 @@
  * @summary fix default handling and other misc
- * @run main/manual Default
+ * @run main/othervm Default
  */
 
 import com.sun.security.auth.callback.TextCallbackHandler;
+import jdk.test.lib.Asserts;
+
 import javax.security.auth.callback.*;
+import java.io.*;
 
 public class Default {
-   public static void main(String args[]) throws Exception {
-        TextCallbackHandler h = new TextCallbackHandler();
-        NameCallback nc = new NameCallback("Name: ", "charlie");
-        ConfirmationCallback cc = new ConfirmationCallback
-                        ("Correct?",
-                        ConfirmationCallback.INFORMATION,
-                        ConfirmationCallback.YES_NO_OPTION,
-                        ConfirmationCallback.NO);
-
-        Callback[] callbacks = { nc, cc };
-        h.handle(callbacks);
-
-        if (cc.getSelectedIndex() == ConfirmationCallback.YES) {
-            System.out.println("yes");
-        } else {
-            System.out.println("no");
+    public static void main(String args[]) throws Exception {
+
+        InputStream in = System.in;
+        PrintStream err = System.err;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        final String defaultName = "charlie";
+        try (PipedInputStream pipedIn = new PipedInputStream();
+             PipedOutputStream pipedOut = new PipedOutputStream(pipedIn);
+             PrintStream prints = new PrintStream(baos)) {
+
+            System.setIn(pipedIn);
+            System.setErr(prints);
+            TextCallbackHandler textHandler = new TextCallbackHandler();
+            NameCallback nameCallback = new NameCallback("Name: ", defaultName);
+            ConfirmationCallback confirmationCallback = new ConfirmationCallback(
+                    "Correct?",
+                    ConfirmationCallback.INFORMATION,
+                    ConfirmationCallback.YES_NO_OPTION,
+                    ConfirmationCallback.NO);
+
+            handleCallback(pipedOut, textHandler, nameCallback);
+            handleCallback(pipedOut, textHandler, confirmationCallback);
+
+            Asserts.assertEquals(nameCallback.getDefaultName(), defaultName);
+            Asserts.assertEquals(confirmationCallback.getSelectedIndex(), ConfirmationCallback.NO);
+
+        } finally {
+            System.setIn(in);
+            System.setErr(err);
         }
-   }
+
+        // check that the default name and confirmation were visible in the output
+        Asserts.assertTrue(baos.toString().contains(String.format("Name:  [%s]", defaultName)));
+        Asserts.assertTrue(baos.toString().contains("1. No [default]"));
+    }
+
+    private static void handleCallback(PipedOutputStream pipedOut, TextCallbackHandler textHandler,
+                                       Callback callback)
+            throws IOException, UnsupportedCallbackException {
+        pipedOut.write("-1\n".getBytes());
+        pipedOut.flush();
+        textHandler.handle(new Callback[]{callback});
```

Comment:
Can you handle 2 `Callback`s in a single `handle` call?

Reply:
Try if this class works for you. https://github.com/openjdk/jdk/blob/45a616a891e4a4b0e77b1f2fa040522f4a99d172/test/jdk/sun/security/tools/keytool/KeyToolTest.java#L1885

Reply:
@wangweij 
I was handling the callback in 2 separate calls because it was blocking and timing out while waiting for input. Now, using _HumanInputStream_ allows handling it in a single call. Unfortunately this class is not visible for this jteg test. To compile it properly, I would need to include the following annotations in the test:
```
 * @modules java.base/sun.security.tools.keytool
 *          java.base/sun.security.util
 *          java.base/sun.security.x509
 * @compile ../../../../../../sun/security/tools/keytool/KeyToolTest.java
```
 But, this would couple both tests together. would it be useful extracting HumanInputStream into a common library class?

Reply:
Sure, you can put it into `test/jdk/java/security/testlibrary`. I'm not sure if it's mature enough to be put into `test/lib`.

Reply:
Ok, it's moved now and updated the tests